operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -675,6 +677,27 @@ public Conditions(DomainStatus status) {
 
         void apply() {
           conditionList.forEach(newCondition -> addCondition(status, newCondition));
+          setClusterStatusConditions(status.getClusters());
+        }
+
+        private void setClusterStatusConditions(List<ClusterStatus> clusterStatuses) {
```

Comment:
Please reverse this logic. Each ClusterCheck contains the name of its cluster, so you iterate over the checks and then ask for the corresponding status. This would let you move the `addClusterConditions` code into ClusterCheck and not need to test if it's null.